pkg/ingester/activeseries/active_series.go

```diff
@@ -100,11 +100,29 @@ func (c *ActiveSeries) purge(keepUntil time.Time) {
 	}
 }
 
-// Active returns the total number of active series, as well as a slice of active series matching each one of the
+func (c *ActiveSeries) ContainsRef(ref uint64) bool {
+	stripeID := ref % numStripes
+	return c.stripes[stripeID].containsRef(ref)
+}
+
+// Active returns the total number of active series. This method does not purge
+// expired entries, so ActiveWithMatchers should still be called periodically.
+func (c *ActiveSeries) Active() int {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```

Comment:
```suggestion
	c.mu.RLock()
	defer c.mu.RUnlock()
```

Reply:
Actually, can you double or triple check whether we need to hold this lock, and what's the purpose of it? As far as I can see, we don't hold it in `UpdateSeries`, so what's the point in holding it here?

I have a feeling that it's more related to a possible matchers change.

Reply:
each stripe also has an individual lock. I also think the global lock on ActiveSeries is only for config. We should probably rename it in that case